src/server/routes/search.js

```diff
@@ -142,23 +144,11 @@ router.get('/exists', (req, res) => {
  *
  * @throws {error.PermissionDeniedError} - Thrown if user is not admin.
  */
-router.get('/reindex', auth.isAuthenticated, (req, res) => {
+router.get('/reindex', auth.isAuthenticated, auth.isAuthorized(REINDEX_SEARCH_SERVER), async (req, res) => {
 	const {orm} = req.app.locals;
-	const indexPromise = new Promise((resolve) => {
-		// TODO: This is hacky, and we should replace it once we switch to SOLR.
-		const trustedUsers = ['Leftmost Cat', 'LordSputnik', 'Monkey', 'iliekcomputers'];
```

Comment:
Thank you !
I've been wanted this gone since the first time I saw it in the codebase !